pkg/steps/template.go

```diff
@@ -564,6 +565,15 @@ func waitForPodCompletion(podClient coreclientset.PodInterface, name string, not
 			continue
 		}
 		if err != nil {
+			if len(artifactDir) > 0 {
+				podList, err := getPodListByName(podClient, name)
+				if err != nil {
+					log.Printf("could not list pod: %v", err)
```

Comment:
Cannot continue to execute when `err != nil`, because then `podList` can be `nil` on line 573 and we will SEGV.